openfeature/openfeature_api.go

```diff
@@ -6,81 +6,70 @@ import (
 	"sync"
 
 	"github.com/go-logr/logr"
-	"github.com/open-feature/go-sdk/openfeature/internal"
 	"golang.org/x/exp/maps"
 )
 
-// evaluationAPI wraps OpenFeature evaluation API functionalities
-type evaluationAPI struct {
+// ofApiImpl is an internal reference interface extending IOFApi
+type ofApiImpl interface {
```

Comment:
I'm not clear on why this intermediate interface is necessary - can you clarify why it exists?

Reply:
I introduced this to support internal work and avoid direct access to singleton instance from other components like `Client`. For example, a client instance needs to obtain the provider associate with its domain, so it rely on `ForEvaluation` method exposed by this internal interface.